addons/account/models/account_move_line.py

```diff
@@ -1989,20 +2045,45 @@ def _prepare_reconciliation_single_partial(self, debit_values, credit_values, sh
                 debit_rate = debit_available_residual_amounts.get(debit_currency, {}).get('rate')
                 credit_rate = credit_available_residual_amounts.get(credit_currency, {}).get('rate')
 
-            # Compute the partial amount expressed in company currency.
-            partial_amount = min_recon_amount
-
-            # Compute the partial amount expressed in foreign currency.
             if debit_rate:
-                partial_debit_amount_currency = debit_currency.round(debit_rate * min_recon_amount)
-                partial_debit_amount_currency = min(partial_debit_amount_currency, remaining_debit_amount_curr)
+                if is_cross_currency_payment:
+                    partial_debit_amount_currency_new = debit_currency.round(remaining_debit_amount * debit_rate)
+                    partial_debit_amount_currency = min(partial_debit_amount_currency_new, min_recon_amount)
+                    if debit_fully_matched:
+                        partial_debit_amount = company_currency.round(min_recon_amount / debit_currency_rate_during_payment)
+                    else:
+                        partial_debit_amount = company_currency.round(min_recon_amount / debit_currency_rate_during_creation)
+                    partial_debit_amount = min(partial_debit_amount, remaining_debit_amount)
```

Comment:
I don't think the `_new` makes a lot of senseif you only put it in that place and regarding the if else block I onelined, i find it better like this and i think i would have written it this way but if you find the line too long which is totally understandable, I'd understand you keep it your way. Nothing major once again.
```suggestion
                if is_cross_currency_payment:
                    partial_debit_amount_currency = debit_currency.round(remaining_debit_amount * debit_rate)
                    partial_debit_amount_currency = min(partial_debit_amount_currency, min_recon_amount)
                    partial_debit_amount = company_currency.round(min_recon_amount / (debit_currency_rate_during_payment if debit_fully_matched else debit_currency_rate_during_creation))
                    partial_debit_amount = min(partial_debit_amount, remaining_debit_amount)
```